### PR TITLE
Sites support

### PR DESCRIPTION
**Pull request by taranvohra**
This PR adds support for rendering sites in GitBook Open.

### CODE REVIEW

File: src/lib/api.ts [hunk 1]
```diff
@@ -834,6 +991,18 @@ export function getAPICacheTag(
...
               syncedBlock: string;
+          }
+        // All data related to a site
+        | {
+              tag: 'site';
+              site: string;
+              organization: string;
+          }
+        | {
+              tag: 'site-space';
```
Comment (SamyPesse) [inline src/lib/api.ts:1002]: What will change with this? should we have everything under one tag `site` ?

File: src/middleware.ts [hunk 2]
```diff
@@ -612,6 +626,9 @@ async function lookupSpaceByAPI(
             apiToken: data.apiToken,
             cacheMaxAge: data.cacheMaxAge,
             cacheTags: data.cacheTags,
+            ...('site' in data ? { site: data.site } : {}),
+            ...('siteSpace' in data ? { siteSpace: data.siteSpace } : {}),
+            ...('organization' in data ? { organization: data.organization } : {}),
```
Comment (SamyPesse) [inline src/middleware.ts:631]: Can't the 3 of these be in just one `...(`?

**Review by SamyPesse — APPROVED** [review]
We should manually test on the preview url a bunch of important GitBook content (with collection, etc) before merging

File: src/app/(space)/(content)/[[...pathname]]/page.tsx [hunk 3]
```diff
@@ -104,13 +104,13 @@ export async function generateViewport({ params }: { params: PagePathParams }):
...
 export async function generateMetadata({ params }: { params: PagePathParams }): Promise<Metadata> {
-    const { space, pages, page, customization, collection } = await fetchPageData(params);
+    const { space, pages, page, customization, parent } = await fetchPageData(params);
     if (!page) {
         notFound();
     }
 
     return {
-        title: [page.title, customization.title ?? space.title, collection?.title]
+        title: [page.title, customization.title ?? space.title, parent?.title]
```
[the remark this answers is not in the document]
Comment (taranvohra) [reply src/app/(space)/(content)/[[...pathname]]/page.tsx:113]: `customization` is the customization settings
`parent` is either a `Collection` or `Site` (both of them have title).

**Comment by taranvohra** [comment]
I have manually tested a bunch of collection URLs that seem to be working

**Pull request merged**